### ISSUE TITLE
Placing fences on any mine floor and then blowing them up spawns a ladder

### ISSUE THREAD

**Issue by JohnsonNicholas** [JohnsonNicholas/USDVP#9]
Most noticeable on floor 120. 

**Description:** If you place fences on any mine floor, and then blow them up, it'll spawn a ladder.

**Source:** Me, the internet, etc.

**Why does this happen?**

It's caused by GameLocation.destroyObject
```cs
if (this is MineShaft && !o.Name.Contains("Lumber"))
      {
        flag = true;
        (this as MineShaft).checkStoneForItems(o.ParentSheetIndex, (int) tileLocation.X, (int) tileLocation.Y, who);
      }
```
where checkStoneForItems is (among other things)
```cs
if (!this.ladderHasSpawned && (this.stonesLeftOnThisLevel == 0 || r.NextDouble() < num3))
        this.createLadderDown(x, y);
```

As a note, the exact cause is

I'd add in the report that num3 > 1 is the cause of creating the ladder i.e. its the rand check that passes the if because Math.Max(1, this.stonesLeftOnThisLevel) == 1

**Fix**

Replace !o.Name.Contains("Lumber") with o.Name == "Stone"

**Note:** As a side note: this only works when there are no other stones.

**Official Fix:** None, reported since 1.0

**Comment by JohnsonNicholas** [comment]
Thanks to Cat, who traced this down before I could!

**Comment by JohnsonNicholas** [comment]
Not fixed in 1.3.12. They just did .Contains. This means the Geode Stone, Stone Fence, Stone Brazier and Stone Sign still work.

**Comment by JohnsonNicholas** [comment]
They fixed it in .14 by excluding craftables. Well.,.. OK.
